pkg/controller/infrastructure/infraflow/ensurer.go

```diff
@@ -632,9 +634,10 @@ func (fctx *FlowContext) ensureSubnets(ctx context.Context) (err error) {
 		return *s.Name
 	})
 	// clean the current inventory and rebuild it.
-	for _, name := range fctx.inventory.ByKind(KindSubnet) {
-		if subnet, ok := mappedSubnets[name]; !ok {
-			fctx.inventory.Delete(*subnet.ID)
+	for _, resource := range fctx.inventory.ByKind(KindSubnet) {
+		if _, ok := mappedSubnets[resource.Name]; !ok {
+			log.V(1).Info("removing subnet from inventory", "id", resource.String())
```

Comment:
Why it is a debug log? IMO it deserves to be a log shown by default because it won't be shown always as some other log entries that will be displayed always on reconciliation like "adding resource to inventory". During investigations this log won't be shown and the log level should be explicitly raised on the extension side via flag.

```suggestion
			log.Info("Removing subnet from inventory", "id", resource)
```